sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/policy/RequestRetryPolicy.java

```diff
@@ -126,7 +112,30 @@ private Mono<HttpResponse> attemptAsync(final HttpPipelineCallContext context, H
 
         // We want to send the request with a given timeout, but we don't want to kick off that timeout-bound operation
         // until after the retry backoff delay, so we call delaySubscription.
-        Mono<HttpResponse> responseMono = next.clone().process();
+        Mono<HttpResponse> responseMono;
+        if (mayTryAgain) {
+            /*
+             * Clone the original request to ensure that each try starts with the original (unmutated) request. We
+             * cannot simply call httpRequest.copy() because although the body will start emitting from the beginning of
+             * the stream, the buffers that were emitted will have already been consumed (their position set to their
+             * limit), so it is not a true reset. By adding the map function, we ensure that anything which consumes the
+             * ByteBuffers downstream will only actually consume a duplicate so the original is preserved. This only
+             * duplicates the ByteBuffer object, not the underlying data.
+             */
+            BinaryData originalRequestBody = originalRequest.getBodyAsBinaryData();
+            if (originalRequestBody != null && !originalRequestBody.isReplayable()) {
+                // Replayable bodies don't require this transformation.
+                // TODO (kasobol-msft) Remove this transformation in favor of
```

Comment:
I assume we can't do it yet, right?

Reply:
Correct, there are some other changes that need to happen to the Flux<ByteBuffer>-based BinaryData to make this safe to use.